packages/backend/src/managers/applicationManager.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { expect, test, vi } from 'vitest';
+import { ApplicationManager } from './applicationManager';
+import type { RecipeStatusRegistry } from '../registries/RecipeStatusRegistry';
+import type { ExtensionContext } from '@podman-desktop/api';
+import type { GitManager } from './gitManager';
+import os from 'os';
+import fs, { type Dirent } from 'fs';
+import path from 'path';
+
+test('appUserDirectory should be under home directory', () => {
+  vi.spyOn(os, 'homedir').mockReturnValue('/home/user');
+  const manager = new ApplicationManager({} as GitManager, {} as RecipeStatusRegistry, {} as ExtensionContext);
+  expect(manager.appUserDirectory).toMatch(/^\/home\/user/);
+});
+
+test('getLocalModels should return models in local directory', () => {
+  vi.spyOn(os, 'homedir').mockReturnValue('/home/user');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  vi.spyOn(fs, 'readdirSync').mockImplementation((dir: string): any => {
+    // TODO(feloy): fix any
```

Comment:
The difficulty with readdirSync is that it returns either Dirent[] or string[] depending on the options passed. I'm not sure how to handle this without using any here...

Reply:
```ts
const readdirSyncMock: MockInstance<[path: fs.PathLike], string[]> = vi.spyOn(fs, 'readdirSync') as unknown as MockInstance<[path: fs.PathLike], string[]>;
readdirSyncMock.mockImplementation((path: fs.PathLike) => {
```

Reply:
(that said on unit test I'm fine to use `any` type)

Reply:
Thanks